packages/core/src/document.ts

```diff
@@ -172,46 +151,87 @@ class Document extends EventEmitter {
    * @param version - Document version
    */
   async loadVersion<T extends Doctype>(version: CID): Promise<T> {
-    const doc = await Document.loadVersion<T>(this, version, this._validate)
+    const doc = await Document.loadVersion<T>(this.id, version, this._doctypeHandler as DoctypeHandler<T>, this.dispatcher, this.pinStore, this._context, this._validate)
     return doc.doctype as T
   }
 
   /**
-   * Loads a specific version of the Doctype
-   *
-   * @param doc - Document instance
-   * @param version - Document version
-   * @param validate - Validate content against schema
+   * Loads a specific version of the document
+   * @param id - base document id
+   * @param version - version of the document being requested
```

Comment:
Can we make version as part of the `id`. Would make more sense since the DocID class includes version anyway. Right now it would be confusing if you have a versioned DocID  and  want to load it.